lib/graphql/metrics/analyzer.rb

```diff
@@ -93,37 +101,88 @@ def result
         case argument
         when Array
           argument.each do |a|
-            extract_arguments(a, field_defn, parent_input_object)
+            extract_arguments(
+              argument: a,
+              definition: definition,
+              parent_input_object: parent_input_object,
+              parent: parent
+            )
           end
         when Hash
           argument.each_value do |a|
-            extract_arguments(a, field_defn, parent_input_object)
+            extract_arguments(
+              argument: a,
+              definition: definition,
+              parent_input_object: parent_input_object,
+              parent: parent)
           end
         when ::GraphQL::Execution::Interpreter::Arguments
           argument.each_value do |arg_val|
-            extract_arguments(arg_val, field_defn, parent_input_object)
+            extract_arguments(
+              argument: arg_val,
+              definition: definition,
+              parent_input_object: parent_input_object,
+              parent: parent
+          )
           end
         when ::GraphQL::Execution::Interpreter::ArgumentValue
-          extract_argument(argument, field_defn, parent_input_object)
-          extract_arguments(argument.value, field_defn, parent_input_object)
+          extract_argument(
+            value: argument,
+            definition: definition,
+            parent_input_object: parent_input_object,
+            parent: parent
+          )
+          extract_arguments(
+            argument:argument.value,
+            definition: definition,
+            parent_input_object: parent_input_object,
+            parent: parent
+          )
         when ::GraphQL::Schema::InputObject
           input_object_argument_values = argument.arguments.argument_values.values
           parent_input_object = input_object_argument_values.first&.definition&.owner
 
-          extract_arguments(input_object_argument_values, field_defn, parent_input_object)
+          extract_arguments(
+            argument: input_object_argument_values,
+            definition: definition,
+            parent_input_object: parent_input_object,
+            parent: parent
+          )
         end
       end
 
-      def extract_argument(value, field_defn, parent_input_object = nil)
+      def extract_argument(value:, definition:, parent_input_object:, parent:)
+        parent_type_name = if definition.is_a?(GraphQL::Schema::Field)
+          definition.owner.graphql_name
+        else
+          DIRECTIVE_TYPE
+        end
+
+        grand_parent_name = if parent.is_a?(GraphQL::Language::Nodes::OperationDefinition)
+          parent.operation_type
+        else
+          parent.name
+        end
+
         static_metrics = {
           argument_name: value.definition.graphql_name,
           argument_type_name: value.definition.type.unwrap.graphql_name,
-          parent_field_name: field_defn.graphql_name,
```

Comment:
As mentioned on Slack, this is a major breaking change, so we'll want to bump the gem to 5.0.0 once this branch/the merge commit are tested against Shopify Core CI.